core/launcher/src/manager/app_launcher.rs

```diff
@@ -513,8 +514,34 @@ impl AppLauncher {
         }
     }
 
-    fn get_transport(url: String) -> AppRuntimeTransport {
-        if !url.as_str().contains("__firebolt_endpoint") {
+    fn get_manifest_url(manifest: AppManifest) -> String {
+        let mut url = manifest.start_page.to_string();
+        if !(url.clone().as_str().contains("__firebolt_endpoint")) {
+            if let Ok(mut parsed_url) = Url::parse(url.clone().as_str()) {
+                let firebolt_endpoint = String::from("127.0.0.0");
+                let appid = manifest.name.to_string();
+
+                let modified_query: String = format!(
+                    "__firebolt_endpoint=ws%3A%2F%2F{}%3A3473%3FappId%3D{}",
+                    firebolt_endpoint, appid
+                );
+                parsed_url.set_query(Some(&modified_query));
```

Comment:
This will override the query for all the url(s). 
For eg if an app url has
```
https://cool.app?neededParam=foo
```

It will change to 
```
https://cool.app?__firebolt_endpoint=ws%3A%2F%2F127.0.0.1%3A3473%3FappId%3D
```
Apps would fail if some mandatory params went missing.
My suggestion is to append to an existing url param before set_query

Reply:
changes added in latest commit.